xbmc/video/VideoDatabase.cpp

```diff
@@ -6019,6 +6217,40 @@ bool CVideoDatabase::GetSetsByWhere(const std::string& strBaseDir, const Filter
   return false;
 }
 
+bool CVideoDatabase::GetTypesNav(const std::string& strBaseDir, CFileItemList& items, int idMedia /* = -1 */)
+{
+  if (m_pDB.get() == nullptr || m_pDS.get() == nullptr)
+    return false;
+
+  try
+  {
+    std::string strSQL;
+
+    if (idMedia != -1)
+      strSQL = PrepareSQL("SELECT type.name, type.type_id FROM type JOIN type_link ON type_link.type_id = type.type_id WHERE media_id = %i", idMedia);
+    else
+      strSQL = PrepareSQL("SELECT name, type_id FROM type");
+
+    m_pDS->query(strSQL);
+
+    while (!m_pDS->eof())
+    {
+      CFileItemPtr pItem(new CFileItem());
```

Comment:
I would declare `pItem` outside the while loop. No need to construct the object on every iteration.